### PR TITLE
Added GROUP syntax

### PR DESCRIPTION
**Pull request by Snooz82**
Implements #5257

### CODE REVIEW

**Comment by Snooz82** [comment]
@pekkaklarck I think i am done.
The one open thing i am aware of is the User Guide.
I would ask you to write that, then i can focus on the certs and the exam question.

Let me know if i shall do changes.

File: src/robot/model/control.py [hunk 1]
```diff
@@ -229,6 +229,37 @@ def __str__(self) -> str:
...
+        self.name = name
+        self.parent = parent
+        self.body = ()
+
+    @setter
+    def body(self, body: 'Sequence[BodyItem|DataDict]') -> Body:
+        return self.body_class(self, body)
+
+    def visit(self, visitor: SuiteVisitor):
+        visitor.visit_while(self)
```
Comment (pekkaklarck) [inline src/robot/model/control.py:251]: Should be `visit_group`. It seems tests for visitor interface are missing.

**Review by pekkaklarck — CHANGES_REQUESTED** [review]
Looks excellent! There we few small things to fix (most importantly `visit_while` -> `visit_group`) that I mentioned in separate comments. In addition to that, these things need to be still handled:

- Investigate why tests fail. They seem to have passed initially, but now after I fixed conflicts with master there are failures. Could be due to changes I've done.
- Add tests for GROUP with templates. This likely requires some further code fixes too. Can be done after merging the PR, but needs to be done before the beta.
- Write documentation. Can be done after beta and even after the rc.

I can take care of fixing these issues, but I won't have time for that right now. If you @Snooz82 want to do something, we can coordinate on Slack.

File: doc/schema/result.xsd [hunk 2]
```diff
@@ -259,6 +265,14 @@
...
         </xs:choice>
     </xs:complexType>
+    <xs:complexType name="Group">
+        <xs:choice maxOccurs="unbounded">
+            <xs:element name="kw" type="Keyword" minOccurs="0" maxOccurs="unbounded" />
+            <xs:element name="msg" type="Message" minOccurs="0" maxOccurs="unbounded" />
+            <xs:element name="status" type="Status" />
+        </xs:choice>
+        <xs:attribute name="name" type="xs:string" />
+    </xs:complexType>
```
[the remark this answers is not in the document]
Comment (pekkaklarck) [reply doc/schema/result.xsd:285]: Resolved.

File: src/robot/parsing/parser/blockparsers.py [hunk 3]
```diff
@@ -44,10 +44,12 @@ def __init__(self, model: Block):
         super().__init__(model)
         self.parsers: 'dict[str, type[NestedBlockParser]]' = {
             Token.FOR: ForParser,
+            Token.GROUP: GroupParser,
             Token.IF: IfParser,
             Token.INLINE_IF: IfParser,
             Token.TRY: TryParser,
             Token.WHILE: WhileParser
+
```
[the remark this answers is not in the document]
Comment (pekkaklarck) [reply src/robot/parsing/parser/blockparsers.py]: Resolved.

File: src/robot/model/control.py (hunk 1 above)
Comment (pekkaklarck) [reply src/robot/model/control.py]: Resolved.

**Comment by pekkaklarck** [comment]
Test failures were due to my changes to master. I've fixed all minor issues that I noticed. Assuming tests pass, I'll merge this and then handle templates afterwards.

**Pull request merged**